crates/bpf-builder/include/loop.bpf.h

```diff
@@ -14,7 +14,7 @@
 // Note: callback_fn must be declared as `static __always_inline` to satisfy the
 // verifier. For some reason, having this double call to the same non-inline
 // function seems to cause issues.
-#ifdef FEATURE_FN_POINTERS
+#ifdef FEATURE_BPF_LOOP
 #define LOOP(max_iterations, max_unroll, callback_fn, ctx)                     \
   if (bpf_core_enum_value_exists(enum bpf_func_id, BPF_FUNC_loop)) {           \
```

Comment:
having this protected by a feature it's not needed to check for the the `bpf_loop`'s `bpf_func_id`

Reply:
True, let me check if it works without.

Reply:
Done

Reply:
what I meant was the opposite.

we had the function pointer feature because the verifier is not able even to load a program with runtime check using `bpf_core_enum_value_exists` because in the `bpf_loop` branch it's using a funcion pointer.

But now that you check for `bpf_loop` before loading the program you can have

```c
#define LOOP(max_iterations, max_unroll, callback_fn, ctx)                     \
  bpf_loop(max_iterations, callback_fn, ctx, 0);   
```
 if there is `FEATURE_BPF_LOOP`

otherwise 

```c
#define LOOP(max_iterations, max_unroll, callback_fn, ctx)                     \
  _Pragma("unroll") for (int i = 0; i < max_unroll; i++) {                     \
    if (callback_fn(i, ctx) == LOOP_STOP)                                      \
      break;                                                                   \
  }
```

in this case you are not depending of fn pointer check for the runtime evaluation in the verifier, but when you detect `bpf_loop` you load the appropriate probe